data/store/mongo/mongo_recorder.go

```diff
@@ -0,0 +1,74 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/tidepool-org/platform/alerts"
+	"github.com/tidepool-org/platform/errors"
+	structuredmongo "github.com/tidepool-org/platform/store/structured/mongo"
+)
+
+// recorderRepo implements RecorderRepository, writing data to a MongoDB collection.
+type recorderRepo structuredmongo.Repository
+
+func (r *recorderRepo) RecordReceivedDeviceData(ctx context.Context,
+	lastComm alerts.LastCommunication) error {
+
+	opts := options.Update().SetUpsert(true)
+	_, err := r.UpdateOne(ctx, r.filter(lastComm), bson.M{"$set": lastComm}, opts)
+	if err != nil {
+		return fmt.Errorf("upserting alerts.LastCommunication: %w", err)
+	}
+	return nil
+}
+
+func (r *recorderRepo) EnsureIndexes() error {
+	repo := structuredmongo.Repository(*r)
+	return (&repo).CreateAllIndexes(context.Background(), []mongo.IndexModel{
+		{
+			Keys: bson.D{
+				{Key: "lastReceivedDeviceData", Value: 1},
+			},
+			Options: options.Index().
+				SetName("LastReceivedDeviceData"),
+		},
+		{
+			Keys: bson.D{
+				{Key: "dataSetId", Value: 1},
+			},
+			Options: options.Index().
+				SetUnique(true).
+				SetName("DataSetIdUnique"),
+		},
+	})
+}
+
+func (r *recorderRepo) filter(lastComm alerts.LastCommunication) map[string]any {
+	return map[string]any{
+		"userId":    lastComm.UserID,
+		"dataSetId": lastComm.DataSetID,
+	}
+}
+
+func (d *recorderRepo) UsersWithoutCommunication(ctx context.Context) ([]alerts.LastCommunication, error) {
+	start := time.Now().Add(-5 * time.Minute)
```

Comment:
Shouldn't this value be passed as a parameter?

Reply:
Could you share your reasoning? I don't see any other value that makes sense.

Any value less than 5 minutes doesn't qualify to send an alert, which is why this value is used. I'll make that more obvious by defining a named constant.

Reply:
`UsersWithoutCommunication` doesn't actually reflect what the implementation does - it only returns users which didn't have communication in the last 5 minutes. `UsersWithoutCommunication(300 * time.Second)` on the other hand reflects this. I see that you renamed this to `OverdueCommunications` which is better.